### DISCUSSION TITLE
`@pnpm` packages default named export vs type default export

### DISCUSSION THREAD

**Discussion by hubertkuoch** [pnpm/pnpm#5459] [Q&A] [answered]
I noticed that all the very useful utilities are usable but I need to append `.default` to make them work and hence I get linting error on package typings.

For example if I want to use `@pnpm/error`:
<img width="545" alt="Screenshot 2022-10-05 at 19 38 58" src="https://user-images.githubusercontent.com/50402245/194125680-953552b3-0478-430e-b29d-eed8e1e70bf7.png">
<img width="642" alt="Screenshot 2022-10-05 at 19 39 12" src="https://user-images.githubusercontent.com/50402245/194125705-dcdec478-0e00-43a1-8440-6268446ab7f0.png">

I think it is because in the `index.d.ts` there is a `default` export...
<img width="465" alt="Screenshot 2022-10-05 at 19 42 01" src="https://user-images.githubusercontent.com/50402245/194126168-828957d9-067c-40f2-b5b0-188049813851.png">
...whereas in the `index.js` file, all exports are named exports, including the `default`.
<img width="373" alt="Screenshot 2022-10-05 at 19 46 02" src="https://user-images.githubusercontent.com/50402245/194127362-b4f6fc14-332b-47be-b0bc-6b85e0e54167.png">

How can we solve those import issues?

**Comment by zkochan** [comment, marked as answer]
I think it would be a good idea to use only named exports. We can set up a linting rule for it.

**Reply by hubertkuoch** [reply]
@zkochan But is there not a mismatch at the @pnpm/error package output level betweeen its index.js named export (exports.default = PnpmError) and its index.d.ts default export (export default type PnpmError = ...)? I feel like so but I see at the same time you use it across the repo as well without any workaround so not sure what's wrong on my side.

**Reply by hubertkuoch** [reply]
Actually I was importing it in a `.mjs` file so it was a babel transpiled esm vs node native esm compat issue
